### DISCUSSION TITLE
How to set dynamic og:image in meta tag base on env or domain name?

### DISCUSSION THREAD

**Discussion by mengtongun** [vercel/next.js#35871] [Help]
How to set dynamic og:image in meta tag base on env or domain name?

**Comment by icyJoseph** [comment]
For the env use `process.env.NODE_ENV`, and match it against, development, production or test.

For the domain name, it is slightly trickier. Servers, don't usually know the domain name of the site they are serving. In fact, white label solution servers might be serving several pages! 

So either, you hard code it at build time, for example Vercel deploys may use `VERCEL_URL`, as per [docs](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables). Other services might offer a similar solution.

Or you can render all pages on demand and check the incoming request to derive the domain name.

Or, you could develop a 3rd party service (or API route perhaps), which through a query, can respond with different images. (Less preferable).

**Reply by mengtongun** [reply]
Could you give more detail ? I seem not so understand.

**Reply by icyJoseph** [reply]
You need to define the domain name as an env variable. A server doesn't know where an application is being hosted. 

If you are deploying in VERCEL, you can activate the VERCEL_URL variable which tells your server in which domain the application is run.

**Reply by mengtongun** [reply]
We have env domain and we set them there.
We want to dynamic open graph image when we sent url to other platform such as google chat....
